p4-16/spec/P4-16-spec.mdk

```diff
@@ -4923,19 +4925,19 @@ packet_in {
 
 ### Fixed width extraction { #sec-packet-extract-one }
 
-The single-argument `extract` method handles fixed-width headers,
+The single-argument `extract` method handles structs or fixed-width headers,
 and is declared in P4 as follows:
 
 ~ Begin P4Example
-void extract<T>(out T headerLeftValue);
+void extract<T>(out T leftValue);
 ~ End P4Example
 
-The expression `headerLeftValue` must evaluate to a l-value (see
-Section [#sec-lvalues]) of type `header` with a fixed width. If
-this method executes successfully, on completion the `headerLvalue`
-is filled with data from the packet and its validity bit is set to `true`. This
+The expression `leftValue` must evaluate to a l-value (see
+Section [#sec-lvalues]) of type `header` with a fixed width, or of type `struct`. If
```

Comment:
A struct can contain any number of nested headers, each of which has may have a varbit.
This is becoming tricky.

Reply:
I guess my comment was wrong - it says here it has to have fixed width.

Reply:
It says elsewhere explicitly that the bit format need not be fixed width.

I can make that more explicit here if you thought the English 'fixed with' modifier extends to cover the struct.  By the insertion of a comma, I was vainly hoping to explicitly separate that.